java/src/main/java/com/saucelabs/saucebindings/SauceOptions.java

```diff
@@ -38,13 +40,27 @@
     private Boolean strictFileInteractability = null;
     private UnhandledPromptBehavior unhandledPromptBehavior;
 
+    // Defined in Appium
+    // These technically belong in Appium's IOSOptions & AndroidOptions
+    // but Sauce has always had them in its documentation, so it should handle them
+    private String app; // TODO - consider handling URL and auto-add "sauce-storage:"
+    private String appActivity; // Android only
```

Comment:
The fact this class includes platform-specific options makes me think it needs to be refactored and split

Reply:
to be more specific

```
sauceOptions.setAppActivity should not be compilable if `sauceOptions` does not target the Android platform
```

Reply:
Yeah, from a CS perspective, this is a "bad class" because not all fields apply to all the Sessions. At the same time, the reason it is bad is because it can theoretically be more difficult to maintain when it isn't abstracted out. The stated goal of the project is to make it easy for our users to get what they want, and when balancing out the added maintenance costs here I believe we should err on the side of usability.

Essentially I don't want users to have to spend time figuring out whether options need to be in `AndroidOptions` or `SauceOptions` when our docs have always included them. If someone needs things that are only specified in Appium docs or tutorials, that's an easier ask to put in a separate class, especially since I expect that to be a small minority of users.

Reply:
I guess I should say that we should be comparing this code to `MutableCapabilities` which has zero protections, and from that standpoint this is better. We could do a better job of evaluating what combinations of capabilities don't work before we send it to Sauce, but I don't think that is MVP for this.